### PR TITLE
feat: fraud prevention messages in Conversations

### PR DESCRIPTION
**Pull request by araujobarret**
The type of this PR is: **Feature**

This PR resolves [NX-3100]

### Description

Changes the Icon and the message about fraud prevention inside a conversation. 
- Removed the block "Be protected..." before the Artwork element in the conversation
- Added a toaster to all messages  with a duration of 5s containing relevant info about transactions inside Artsy
- Changed the Make Offer on Inquiry messages above the CTA button

@artsy/negotiate-devs 

https://user-images.githubusercontent.com/15792853/158141277-0c3bcf2f-8a6e-4e38-9faa-85dc97119e2c.mov

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [ ] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- feat: fraud messages improvement inside conversations - araujobarret

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[NX-3100]: https://artsyproduct.atlassian.net/browse/NX-3100?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (feat: fraud messages improvement inside conversations - araujobarret)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 5985f33279ba029d9b108c406e34bb36a393927d
</p>

File: src/app/Components/Toast/ToastComponent.tsx [hunk 1]
```diff
@@ -14,7 +14,10 @@ const EDGE_TOAST_HEIGHT = 60
 const EDGE_TOAST_PADDING = 10
 const NAVBAR_HEIGHT = 44
 
-type ToastProps = ToastDetails
+type ToastProps = ToastDetails & {
+  backgroundColor?: Color
```
Comment (araujobarret) [inline src/app/Components/Toast/ToastComponent.tsx:18]: The toaster here doesn't support all types and seems to be doing much as needed right now. Just extended some props to keep moving but gonna ticket to refactor our toaster to support all types and be less-batteries-included, this component is doing too much imo.

File: src/app/Scenes/Inbox/Components/Conversations/MessageGroup.tsx [hunk 2]
```diff
@@ -122,7 +105,9 @@ export class MessageGroup extends React.Component<MessageGroupProps> {
       return (
         <>
           <View>
-            <TimeSince style={{ alignSelf: "center" }} time={firstItem.createdAt} exact mb={1} />
+            {!!!isLastMessage && (
```
Comment (araujobarret) [inline src/app/Scenes/Inbox/Components/Conversations/MessageGroup.tsx:108]: to remove a double rendered time info in the message

File: src/app/Scenes/Inbox/Components/Conversations/Messages.tsx [hunk 3]
```diff
@@ -143,34 +145,43 @@ export const Messages: React.FC<Props> = forwardRef((props, ref) => {
...
-      key={conversation.internalID!}
-      data={messages}
-      initialNumToRender={messages?.length}
-      renderItem={({ item, index }) => {
-        return (
-          <>
+    <>
+      <ToastComponent
+        id={1}
+        positionIndex={-0.3}
```
Comment (araujobarret) [inline src/app/Scenes/Inbox/Components/Conversations/Messages.tsx:151]: this is ugly but was the only way to make the toaster stick to the desired position, we should refactor it in the Toaster update ticket.

File: src/app/Components/Toast/ToastComponent.tsx (hunk 1 above)
Comment (jo-rs) [reply src/app/Components/Toast/ToastComponent.tsx:18]: Nice nice

File: src/app/Components/Toast/ToastComponent.tsx [hunk 4]
```diff
@@ -64,10 +69,11 @@ export const ToastComponent: React.FC<ToastProps> = ({
         position="absolute"
         top={(height - MIDDLE_TOAST_SIZE) / 2}
         left={(width - MIDDLE_TOAST_SIZE) / 2}
-        backgroundColor={color("black100")}
+        backgroundColor={color(backgroundColor)}
         opacity={opacityAnim.interpolate({ inputRange: [0, 1], outputRange: [0, 0.9] })}
-        borderRadius={5}
```
Comment (jo-rs) [inline src/app/Components/Toast/ToastComponent.tsx:69]: Was removing this defined for all uses of the toast? 🤔

File: src/app/Components/Toast/ToastComponent.tsx [hunk 5]
```diff
@@ -64,10 +69,11 @@ export const ToastComponent: React.FC<ToastProps> = ({
...
         overflow="hidden"
+        paddingRight={2}
+        zIndex={999}
```
Comment (jo-rs) [inline src/app/Components/Toast/ToastComponent.tsx:76]: I suspect that fundamentally it shouldn't, but could this be an issue somewhere else? 🤔

File: src/app/Components/Toast/types.ts [hunk 6]
```diff
@@ -19,4 +19,4 @@ export interface ToastDetails {
   Icon?: React.FC<IconProps>
 }
 
-export type ToastOptions = Omit<ToastDetails, "id" | "positionIndex" | "placement" | "message">
+export type ToastOptions = Pick<ToastDetails, "onPress" | "Icon">
```
Comment (jo-rs) [inline src/app/Components/Toast/types.ts:22]: Nice. A lot cleaner 🧹

File: src/app/Scenes/Inbox/Components/Conversations/MessageGroup.tsx (hunk 2 above)
Comment (jo-rs) [reply src/app/Scenes/Inbox/Components/Conversations/MessageGroup.tsx:108]: Do we generally use `!!!` vs. `!`? 🤔

File: src/app/Scenes/Inbox/Components/Conversations/Messages.tsx (hunk 3 above)
Comment (jo-rs) [reply src/app/Scenes/Inbox/Components/Conversations/Messages.tsx:151]: Hmm I wonder about the consistency of this 🤔
Will try checking in a bit.

**Review by jo-rs — DISMISSED** [review]
Looks good 💯 
I just left a few questions and will be testing it locally in a sec.

File: src/app/Components/Toast/ToastComponent.tsx (hunk 5 above)
Comment (araujobarret) [reply src/app/Components/Toast/ToastComponent.tsx:76]: The issue is the component is not supporting the prop drilling from styled-system, I checked in one/two places and looked okayish

**Comment by araujobarret** [comment]
@jo-rs fixed the !!! issue in this [commit](https://github.com/artsy/eigen/pull/6417/commits/0f12eeedf323a7b6ff46c94eca348b62f8617595), ty!

**Pull request merged**